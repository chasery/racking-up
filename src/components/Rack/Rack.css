.Rack {
  list-style: none;
  box-shadow: 0 0.0625rem 0.0625rem #5e43521c;
  border-radius: 0.125rem;
  overflow: hidden;
  background-color: #fffdff;
}
.Rack__header {
  display: flex;
  align-items: center;
  padding: 0 1.125rem;
  background-color: #fffdff;
  font-weight: 500;
  transition: background-color 0.25s ease;
  text-decoration: none;
}
.Rack__header:hover {
  background-color: #ead7dd;
  box-shadow: inset 0 0.0625rem 0.0625rem #5e43521c;
}
.Rack__header > * {
  padding: 1rem 0.375rem;
  color: #5e4352;
  text-decoration: none;
}
.Rack__header > *:hover {
  color: #5e4352;
}
.Rack__headerName {
  flex-grow: 1;
}
.Rack__headerArrow {
  padding: 0 0.375rem;
}
.Rack__headerArrow img {
  margin-top: 0.5rem;
}
.Rack__headerArrow.forward img {
  transform: rotate(180deg);
}

.Rack__headerControls {
  display: flex;
  align-items: center;
}
.Rack__headerControls button {
  background-color: transparent;
  border: none;
}
.Rack__headerControls button + button {
  margin-left: 0.375rem;
}

.Rack__items {
  padding: 0.5rem 1.125rem;
  border-top: 0.0625rem solid #ead7dd;
}
