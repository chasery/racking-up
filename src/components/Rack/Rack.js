import React from "react";
import { useParams, Link } from "react-router-dom";
import RackItem from "../RackItem/RackItem";
import { currencyFormat } from "../../currencyFormat";
import "./Rack.css";

const clothingArr = [
  {
    id: 1,
    rackId: 1,
    name: "H&M Navy Hoody",
    url: "http://localhost:3000/racks",
    price: 50.0,
  },
  {
    id: 2,
    rackId: 1,
    name: "Columbia Rain Jacket",
    url: "http://localhost:3000/racks",
    price: 75.0,
  },
  {
    id: 3,
    rackId: 1,
    name: "Mountain Hardware Long Sleeve Shirt",
    url: "http://localhost:3000/racks",
    price: 25.0,
  },
  {
    id: 4,
    rackId: 1,
    name: "APC Raw Denim Jeans",
    url: "http://localhost:3000/racks",
    price: 200.0,
  },
  {
    id: 5,
    rackId: 2,
    name: "H&M V Neck Tshirts",
    url: "http://localhost:3000/racks",
    price: 25.0,
  },
  {
    id: 6,
    rackId: 2,
    name: "Express Oxford Shirt",
    url: "http://localhost:3000/racks",
    price: 25.0,
  },
  {
    id: 7,
    rackId: 2,
    name: "Prana Dock Shorts",
    url: "http://localhost:3000/racks",
    price: 75.0,
  },
];

const createRackItems = (items) => {
  return items.map((item) => (
    <RackItem
      key={item.id}
      id={item.id}
      rackId={item.rackId}
      name={item.name}
      url={item.url}
      price={item.price}
    />
  ));
};

const getRackCost = (items) => {
  let cost = 0;
  items.map((item) => (cost += item.price));

  return currencyFormat.format(cost);
};

function Rack(props) {
  const { rackId } = useParams();
  const { id, name } = props;
  const rackItems = clothingArr.filter((item) => item.rackId === props.id);

  return (
    <li className="Rack">
      {rackId ? (
        <div className="Rack__header">
          <Link to={`/racks`} className="Rack__headerArrow">
            {"<-"}
          </Link>
          <span className="Rack__headerName">{name}</span>
          <span className="Rack__headerCost">{getRackCost(rackItems)}</span>
          <div className="Rack__headerControls">
            <button>E</button>
            <button>D</button>
          </div>
        </div>
      ) : (
        <Link to={`/racks/${id}`} className="Rack__header">
          <span className="Rack__headerName">{name}</span>
          <span className="Rack__headerCost">{getRackCost(rackItems)}</span>
          <span className="Rack__headerArrow">{"->"}</span>
        </Link>
      )}
      <ul className="Rack__items">
        {rackItems.length ? createRackItems(rackItems) : <RackItem />}
      </ul>
    </li>
  );
}

export default Rack;
