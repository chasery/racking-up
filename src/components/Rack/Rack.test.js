import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Rack from "./Rack";

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"), // use actual for all non-hook parts
  useParams: () => ({
    rackId: "",
  }),
}));

describe("Rack component", () => {
  it("renders without error", () => {
    const wrapper = shallow(<Rack />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
