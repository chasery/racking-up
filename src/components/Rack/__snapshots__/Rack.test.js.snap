// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rack component renders without error 1`] = `
<li
  className="Rack"
>
  <Link
    className="Rack__header"
    to="/racks/undefined"
  >
    <span
      className="Rack__headerName"
    />
    <span
      className="Rack__headerCost"
    >
      $0
    </span>
    <span
      className="Rack__headerArrow"
    >
      -&gt;
    </span>
  </Link>
  <ul
    className="Rack__items"
  >
    <RackItem />
  </ul>
</li>
`;
